lld/COFF/Config.h

```diff
@@ -130,9 +130,9 @@ struct Configuration {
   bool forceMultipleRes = false;
   bool forceUnresolved = false;
   bool debug = false;
-  bool debugDwarf = false;
+  bool includeDwarfChunks = false;
```

Comment:
While discarding DWARF sections (`.debug_*`) was and is the default for lld-link, it's not the default for mingw, is it? I think the "obvious", non-format-aware thing for the linker to do is to treat `.debug_*` sections as normal sections.

Perhaps it would be more straightforward to invert the sense of the boolean, rename this to `discardDwarfChunks`, default it to false, and have the driver set it to true in non-mingw mode to preserve the existing behavior of lld-link. And then the question becomes, is that even the right behavior anymore? I think this default existed to defend against incorrectly emitted DWARF, but I'm not sure we really need it. I believe MSVC links debug info normally, although it truncates the section names.

Feel free to proceed with incremental patches, this is just an idea. If we can make these changes, then we can remove many of the `includeDwarfChunks` flag updates when parsing `/debug:`.

Reply:
Yes, MSVC link.exe just includes any `.debug_whatever` sections as is regardless of `/debug`.

Indeed, for mingw mode, the default is to include any `.debug_*` sections, but it can be overridden if linking with `-S` or `-s` (`--strip-debug` or `--strip-all`). But I'm not really keen on changing this behaviour here on the lld-link level, this is already handled quite nicely by the mingw frontend passing in `/debug:dwarf` or `/debug:symtab` as needed.

I don't think we gain anything really from changing the default of `includeDwarfChunks` to true for mingw mode though. If it would be good for clarity, we could of course invert the meaning of the variable from `includeDwarfChunks` to `discardDwarfChunks`, but that would just be an internal variable rename and negation.

In the end, from the mingw frontend, we need to be able to flag whether to include or dicard dwarf debug info, separately from whether to make a PDB.